lib/openshift/cluster_operator.rb

```diff
@@ -4,12 +4,22 @@ class ClusterOperator < ClusterResource
     # return the an array of versions for a particular clusteroperator
     def versions(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
-      rr.dig('status', 'versions').map { |v| v['version'] }
+      rr.dig('status', 'versions').map { |v| v['version'] if v['name'] == 'operator' }.compact
     end
-    # given a target_version return true if has matching verison else false
-    def version_exists?(version: nil, user: nil, cached: true, quiet: false)
+    # given a target_version return true if has matching version else false
+    def version_exists?(version:, user: nil, cached: true, quiet: false)
       res = self.versions.select { |v| v == version }
       res.count != 0
     end
+
+    # return Hash of conditions for easier access
+    def conditions(user: nil, cached: true, quiet: false)
+      conditions = raw_resource(user: user, cached: cached, quiet: quiet).dig('status', 'conditions')
+      cond_hash = {}
+      conditions.each do |con|
+        cond_hash[con['type']] = con
+      end
+      cond_hash
+    end
```

Comment:
This method can go to `resource.rb` IMO.

Reply:
@akostadinov should've checked it earlier...there's already one [there](https://github.com/openshift/verification-tests/blob/master/lib/openshift/resource.rb#L283), but it just return the raw array of conditions.

Reply:
You can create a method to return to you based on type. It is a big anti-pattern to override a method changing its contract (in this case return data structure). See [Liskov principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

Reply:
e.g. `oc.conditions_by_type("my_type", user: admin)`

Reply:
@akostadinov apparently there's also an existing method `[condition](https://github.com/openshift/verification-tests/blob/master/lib/openshift/resource.rb#L290)`

Reply:
great find!